storage/storage.go

```diff
@@ -63,3 +94,26 @@ func StoreResult(result string, consumerURL string) error {
 
 	return nil
 }
+
+// createReport returns a pointer to a types.Report built from the supplied
+// map of container IDs to types.ContainerReport.
+func createReport(results map[string]types.ContainerReport) (types.Report, error) {
+	logging.Stdout("[Storage] Marshalling JSON")
+	marshalledResult, err := json.Marshal(results)
+	if err != nil {
+		return types.Report{}, err
+	}
+	logging.Stdout("[Storage] Marshalling successful %s", string(marshalledResult))
+
+	report := types.Report{
+		Schema:        "http://puppet.com/lumogon/core/draft-01/schema#1",
+		Generated:     time.Now().String(),
+		Owner:         "default",
+		Group:         []string{"default"},
+		ClientVersion: version.Version,
+		ReportID:      utils.GenerateUUID4(),
+		Containers:    results,
+	}
+	logging.Stdout("[Storage] Report created")
+	return report, nil //TODO do we really want a pointer here?
```

Comment:
re: TODO... my take is no. We might need to in the future, but I think we should wait until profiling tools point to this as a bottleneck on copy or whatnot when the report becomes gigantic. This also may be a reflection point where we can ask ourselves if we need to adjust the report to avoid it becoming too large.